powerscale/provider/network_setting_resource.go

```diff
@@ -0,0 +1,256 @@
+			"We can Create, Update and Delete the Network Settings using this resource. We can also import an existing Network Settings from PowerScale array. " +
+			"Note that, powerscale_network_settings is the native functionality of PowerScale. When creating the resource, we actually load Network Settings from PowerScale to the resource state. ",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Network Settings ID.",
+				MarkdownDescription: "Network Settings ID.",
+				Computed:            true,
+			},
+			"default_groupnet": schema.StringAttribute{
+				Description:         "Default client-side DNS settings for non-multitenancy aware programs.",
+				MarkdownDescription: "Default client-side DNS settings for non-multitenancy aware programs.",
+				Computed:            true,
+			},
+			"source_based_routing_enabled": schema.BoolAttribute{
+				Description:         "Enable or disable Source Based Routing. (Update Supported)",
+				MarkdownDescription: "Enable or disable Source Based Routing. (Update Supported)",
+				Optional:            true,
+				Computed:            true,
+			},
+			"sc_rebalance_delay": schema.Int64Attribute{
+				Description:         "Delay in seconds for IP rebalance. (Update Supported)",
+				MarkdownDescription: "Delay in seconds for IP rebalance. (Update Supported)",
+				Optional:            true,
+				Computed:            true,
+				Validators:          []validator.Int64{int64validator.Between(0, 10)},
+			},
+			"tcp_ports": schema.ListAttribute{
+				Description:         "List of client TCP ports. (Update Supported)",
+				MarkdownDescription: "List of client TCP ports. (Update Supported)",
+				ElementType:         types.Int64Type,
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.List{
+					listvalidator.UniqueValues(),
+					listvalidator.ValueInt64sAre(int64validator.Between(0, 65535)),
+					listvalidator.SizeBetween(0, 65535),
+				},
+			},
+		},
+	}
+}
+
+// Configure configures the resource.
+func (r *NetworkSettingResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pscaleClient, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = pscaleClient
+}
+
+// Create allocates the resource.
+func (r *NetworkSettingResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "Creating Network Settings resource state")
+	var state models.NetworkSettingModel
+	settingResponse, err := helper.GetNetworkSetting(ctx, r.client)
```

Comment:
For settings resource, I've thought about it for a while. For developer, we need to distinguish Read, Create, Update, Delete, so for create, it looks like we should only load settings at create method. But for customer, and HCL perspective, HCL is a declarative language, that means, customer just uses HCL to describe the result the want. When there is no state, and customer specifies some settings arguments, it would make more sense to "create" the settings to align with customer's expectation. In other words, the final state should be what customer needed. 
To achieve that, we need to create/load settings then update settings if HCL arguments are not empty.

Reply:
you mean do create and update at the same time?

Reply:
@baoy1 you mean do create and update at the same time?

Reply:
yes

Reply:
good point, will improve the create action.

Reply:
@baoy1  updated